meecrowave-doc/src/main/jbake/content/howto.adoc

```diff
@@ -0,0 +1,303 @@
+import org.apache.meecrowave.Meecrowave;
+
+/**
+ * A bit of glue to set proxy / RewriteValve configuration at startup
+ * 
+ */
+@Log4j2
+public class RewriteValveCustomizer implements Meecrowave.ConfigurationCustomizer {
+    final String PROXY_CONFIG = "rewrite.config";
+    @Override
+    public void accept(final Meecrowave.Builder builder) {
+        log.info("Loading proxy / rewrite configuration from {}", PROXY_CONFIG);
+        log.info("This file should be in src/main/resources in project sources");
+        try (InputStream stream = Thread.currentThread().getContextClassLoader().getResourceAsStream(PROXY_CONFIG)) {
+            if (null == stream) {
+                log.info("Rewrite configuration file {} not found", PROXY_CONFIG);
+                return;
+            }
+            configuration = new BufferedReader(new InputStreamReader(stream)).lines().collect(Collectors.joining("\n"));
+        } catch (IOException ex) {
+            log.error("Error reading rewrite / proxy configuration file {}", PROXY_CONFIG);
+            return;
+        }
+        final RewriteValve proxy = new RewriteValve() {
+            @Override
+            protected synchronized void startInternal() throws LifecycleException {
+                super.startInternal();
+                try {
+                    setConfiguration(configuration);
+                } catch (final Exception e) {
+                    throw new LifecycleException(e);
+                }
+            }
+        };
+        // at this time, we are still single threaded. So, this should be safe.
+        builder.instanceCustomizer(tomcat -> tomcat.getHost().getPipeline().addValve(proxy));
+        log.info("Proxy / rewrite configuration valve configured and added to tomcat.");
+    }
+}
+----
+
+Your implementation of Meecrowave.ConfigurationCustomizer should be added to the configuration by appending its canonical name to the src/main/resources/META-INF/org.apache.meecrowave.Meecrowave$ConfigurationCustomizer file.
+
+
+A more complex example link:https://rmannibucau.metawerx.net/post/tomcat-rewrite-url[is available on Romain Manni-Bucau's blog].
+
+== How to add a web frontend ?
+
+You should add a <webapp> element to the meecrowave plugin configuration. Example :
+[source,pom.xml]
+----
+<plugin>
+    <!--
+        For starting meecrowave via Maven. Just run
+        $> mvn clean install meecrowave:run
+    -->
+    <groupId>org.apache.meecrowave</groupId>
+    <artifactId>meecrowave-maven-plugin</artifactId>
+    <version>1.2.5</version>
+    <configuration>
+        <!-- include packaged app as webapp -->
+        <webapp>src/main/webapp/dist</webapp>
+    </configuration>
+</plugin>
+----
+
+will add the content of the "dist" folder to your package and its files will be available on the application root.
+
+Note that your frontend will be served when executing the app (on a mvn meecrowave:run or when running a packaged app). It will not be available during unit tests.
```

Comment:
If you can explain how it can be configured in tests, it might be useful to others, I think.
However, in dev, you tend to use some other tools. With vue.js, for instance, you do a "yarn serve" to enable live rebuild of your code on modifications.

Reply:
Think you can just build your front build output in target/classes/META-INF/resources, will enable tests to work OOTB, runtime to work and you can get livereload if you disable caching and set webapp dir to this one, no? Not sure the last point is that useful, front dev often proxy the back - thinking to webpack dev server to not cite it.

Reply:
Yep, it is the same with popular frameworks like angular or vue.js... Not that useful.